src/pretalx/cfp/flow.py

```diff
@@ -339,12 +343,26 @@ def get_form_initial(self):
                         result[field] = obj
         return result
 
+    @context
+    def questions_form(self):
+        return QuestionsForm(
+            data=self.request.POST if self.request.method == "POST" else None,
+            files=self.request.FILES if self.request.method == "POST" else None,
+            event=self.request.event,
+            speaker=self.request.user,
+            target="submission",
+        )
+
     def done(self, request, draft=False):
         self.request = request
         form = self.get_form(from_storage=True)
+        form.speaker = request.user
         form.instance.event = self.event
-        form.save()
+        form.is_valid()
         submission = form.instance
+        form.submission = submission
```

Comment:
I don't understand, what is the point of re-attaching `instance` to form under another name?

This line is 
```py
form.submission = form.instance
```